index.js

```diff
@@ -1,17 +1,139 @@
 const express = require('express');
 const bodyParser = require('body-parser');
-const helmController = require('./autom8s/helm-controller');
-const portController = require('./autom8s/port-controller');
-const ingressController = require('./autom8s/ingress-controller');
+const Helm = require('./autom8s/helm');
+const PortsAllocator = require('./autom8s/ports-allocator');
+const IngressManager = require('./autom8s/ingress-manager');
 
 const app = express();
 
 app.use(bodyParser.json());
 app.use(bodyParser.urlencoded({ extended: false }));
 
-app.use('/', helmController);
-app.use('/', portController);
-app.use('/', ingressController);
+// Helm functionallity
+
+/**
+ * Installs the requested chart into the Kubernetes cluster
+ */
+app.post('/install',
```

Comment:
BTW, you could still keep the logic in the port-controller, ingress-controller objects, etc, and only extract the http service related functionality under app.post, app.get, etc.

My comment was about making sure the api is clear from going over the index.js file without having to go over 4 different files.